src/main/java/fi/dy/masa/malilib/gui/StringListSelectionScreen.java

```diff
@@ -50,6 +50,7 @@ protected void updateWidgetPositions()
     protected void onConfirm()
     {
         this.consumer.accept(this.getListWidget().getSelectedEntries());
+        this.closeScreenOrShowParent();
```

Comment:
This shouldn't be done here, as it would cause any screen the Consumer had already opened to be closed again immediately. The Consumer should be left to close it if needed, or in some cases it would have probably opened some other screen already at this point (like with the Material List screen in Litematica: https://github.com/maruohon/litematica/blob/644dc94978b3900eaa39964097cac092402df6b0/src/main/java/fi/dy/masa/litematica/materials/MaterialListUtils.java#L164-L183).